src/OrchardCore.Modules/OrchardCore.Media.Azure/Startup.cs

```diff
@@ -152,4 +155,64 @@ private static bool CheckOptions(string connectionString, string containerName,
             return optionsAreValid;
         }
     }
+
+    [Feature("OrchardCore.Media.Azure.ImageSharpImageCache")]
+    public class ImageSharpAzureBlobCacheStartup : Modules.StartupBase
+    {
+        private readonly IShellConfiguration _configuration;
+        private readonly ILogger _logger;
+
+        public ImageSharpAzureBlobCacheStartup(IShellConfiguration configuration, ILogger<ImageSharpAzureBlobCacheStartup> logger)
+        {
+            _configuration = configuration;
+            _logger = logger;
+        }
+
+        // This is needed to be greater than OrchardCore.Media's 0 to replace the IImageCache implementation registered
+        // there.
+        public override int Order => 5;
+
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.AddTransient<IConfigureOptions<ImageSharpImageCacheOptions>, ImageSharpImageCacheOptionsConfiguration>();
+            services.AddTransient<IConfigureOptions<AzureBlobStorageCacheOptions>, AzureBlobStorageCacheOptionsConfiguration>();
+
+            // Only replace default implementation if options are valid.
+            var connectionString = _configuration[$"OrchardCore_Media_Azure_ImageSharp_Cache:{nameof(ImageSharpImageCacheOptions.ConnectionString)}"];
```

Comment:
You may want to use _configuration.GetSection("").Get<AzureBlobStorageCacheOptions>(). cleaner